core/internal/filetransfer/file_transfers.go

```diff
@@ -28,10 +28,15 @@ func NewFileTransfers(
 	logger *observability.CoreLogger,
 	fileTransferStats FileTransferStats,
 ) *FileTransfers {
+	filetransfers := &FileTransfers{}
+
 	defaultFileTransfer := NewDefaultFileTransfer(client, logger, fileTransferStats)
-	gcsFileTransfer := NewGCSFileTransfer(nil, logger, fileTransferStats)
-	return &FileTransfers{
-		Default: defaultFileTransfer,
-		GCS:     gcsFileTransfer,
+	filetransfers.Default = defaultFileTransfer
+
+	gcsFileTransfer, err := NewGCSFileTransfer(nil, logger, fileTransferStats)
+	if err == nil {
```

Comment:
it might have issues since we are expecting a GCS file transfer in `task_ref_art.go` in `getStorageProvider` but since we didn't catch error here [`fts.GCS`](https://github.com/wandb/wandb/blob/main/core/internal/filetransfer/task_ref_art.go#L94) might just be `nil`?  and then when you go do [`ft.download()`](https://github.com/wandb/wandb/blob/main/core/internal/filetransfer/task_ref_art.go#L74) it might freak out since ft is nil?

Reply:
yup totally agree, the error was very unclear. I added a check in `ReferenceArtifactTask.getStorageProvider` to check whether `fts.GCS` is nil before returning it, so now it returns an error saying we couldn't instantiate the client if it is nil instead of freaking out, and I added a `CaptureError` in the function above so that we can actually figure out what that error was if we need to